sys/net/application_layer/cord/epsim/cord_epsim.c

```diff
@@ -33,18 +34,23 @@
 static coap_pkt_t pkt;
 static uint8_t buf[BUFSIZE];
 
-int cord_epsim_register(void)
+/* keep state of the latest registration attempt */
+static int _state = CORD_EPSIM_ERROR;
+
+static void _req_handler(unsigned req_state, coap_pkt_t* pdu,
+                         sock_udp_ep_t *remote)
+{
+    (void)remote;
+    (void)pdu;
+    _state = (req_state == GCOAP_MEMO_RESP) ? CORD_EPSIM_OK : CORD_EPSIM_ERROR;
+}
+
+int cord_epsim_register(const sock_udp_ep_t *rd_ep)
```

Comment:
Presently this function just returns a state value. However, before returning it must first *set* the _state variable.

Reply:
fixed